src/escpos/config.py

```diff
@@ -45,13 +46,40 @@ def _reset_config(self) -> None:
         self._printer_name = None
         self._printer_config = None
 
-    def load(self, config_path=None):
+    def _set_config(self, config) -> None:
+        """Set configuration variables.
+
+        :param config: config object loaded from file or string.
+        """
+        self._printer_config = config.get("printer")
+        if self._printer_config is None:
+            self._has_loaded = True
+            return
+        printer_name = self._printer_config.pop("type")
```

Comment:
This passes the tests battery.

I replaced `pop()` with `get()` because a I can't find a reason to remove the "type" key given that there is a check for the existence of a `printer_name` attribute in the printer class later. Anyway, I think that `pop()` could be restored cleanly if needed.

```suggestion
        self._printer_config = config.get("printer", None)
        printer_name = ""
        if self._printer_config:
            printer_name = self._printer_config.get("type", "")
```

After this change we only have to please `mypy` by replacing the initialization and restoration of the `self._printer_name` property from `None` to an empty string:

```python
self._printer_name = ""
```

Reply:
> Currently, we'd need to check for the KeyError

After this change, it is no longer needed.

Reply:
It seems like we need the early return to maintain the behavior where a completely missing printer section raises `ConfigSectionMissingError` inside `printer()`

```python
        if not self._printer_name:
            raise exceptions.ConfigSectionMissingError("printer")
```

I propose we have both 

```python
     def __init__(self) -> None:
        # ...
        self._printer_name = ""   
```

```python
        self._printer_config = config.get("printer", None)
        if self._printer_config is None:
            self._has_loaded = True
            return
        printer_name = self._printer_config.get("type", "")
```

and test for both

```python
def test_config_load_with_empty_printer_section() -> None:
    """Config with an empty printer section should raise ConfigSyntaxError."""
    from escpos import config

    c = config.Config()
    with pytest.raises(escpos.exceptions.ConfigSyntaxError):
        c.load_yaml_string("printer: {}")

def test_config_load_yaml_string_without_printer_section() -> None:
    """Config without a printer section should raise ConfigSectionMissingError."""
    from escpos import config

    c = config.Config()
    c.load_yaml_string("{}")
    with pytest.raises(escpos.exceptions.ConfigSectionMissingError):
        c.printer()
```

Reply:
Looks solid. This is the new suggestion based on your comment:

```suggestion
        self._printer_config = config.get("printer", None)
        if self._printer_config is None:
            self._has_loaded = True
            return
        printer_name = self._printer_config.get("type", "")
```
 Please, modify `self._printer_name = None` to `self._printer_name = ""` in `__init__()` and `_reset_config()` as I can't do it. :love_you_gesture:

Reply:
Thanks for your help @belono.

Reply:
Done in [df166d2](https://github.com/python-escpos/python-escpos/pull/707/commits/df166d2cb9a48c7b1432e6a7ef438385c5650ae6) and [a8ebebb](https://github.com/python-escpos/python-escpos/pull/707/commits/a8ebebbc93549e6f303b4e3cf5dea1e6f1df3281)